src/backend/distributed/commands/view.c

```diff
@@ -0,0 +1,418 @@
+static void AppendOptionsToCreateViewCommand(StringInfo createViewCommand, Oid viewOid);
+static StringInfo CreateAlterViewOwnerCommand(Oid viewOid);
+
+/*
+ * PreprocessViewStmt is called during the planning phase for CREATE OR REPLACE VIEW
+ * before it is created on the local node internally.
+ */
+List *
+PreprocessViewStmt(Node *node, const char *queryString,
+				   ProcessUtilityContext processUtilityContext)
+{
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	/* check creation against multi-statement transaction policy */
+	if (!ShouldPropagateCreateInCoordinatedTransction())
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	return NIL;
+}
+
+
+/*
+ * PostprocessViewStmt actually creates the commmands we need to run on workers to
+ * propagate views.
+ *
+ * If view depends on any undistributable object, Citus can not distribute it. In order to
+ * not to prevent users from creating local views on the coordinator WARNING message will
+ * be sent to the customer about the case instead of erroring out. If no worker nodes exist
+ * at all, view will be created locally without any WARNING message.
+ *
+ * Besides creating the plan we also make sure all (new) dependencies of the view are
+ * created on all nodes.
+ */
+List *
+PostprocessViewStmt(Node *node, const char *queryString)
+{
+	ViewStmt *stmt = castNode(ViewStmt, node);
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	/* check creation against multi-statement transaction policy */
+	if (!ShouldPropagateCreateInCoordinatedTransction())
+	{
+		return NIL;
+	}
+
+	ObjectAddress viewAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+
+	if (IsObjectAddressOwnedByExtension(&viewAddress, NULL))
+	{
+		return NIL;
+	}
+
+	/* If the view has any unsupported dependency, create it locally */
+	DeferredErrorMessage *errMsg = DeferErrorIfHasUnsupportedDependency(&viewAddress);
+
+	if (errMsg != NULL)
+	{
+		/*
+		 * Don't need to give any warning/error messages if there is no worker nodes in
+		 * the cluster as user's experience won't be affected on the single node even
+		 * if the view won't be distributed.
+		 */
+		if (!HasAnyNodes())
+		{
+			return NIL;
+		}
+
+		/*
+		 * If the view is already distributed, we should provide an error to not have
+		 * different definition of view on coordinator and worker nodes. If the view
+		 * is not distributed yet, we can create it locally to not affect user's local
+		 * usage experience.
+		 */
+		if (IsObjectDistributed(&viewAddress))
+		{
+			RaiseDeferredError(errMsg, ERROR);
+		}
+		else
+		{
+			RaiseDeferredError(errMsg, WARNING);
+			return NIL;
+		}
+	}
+
+	EnsureDependenciesExistOnAllNodes(&viewAddress);
+
+	List *commands = list_make1(DISABLE_DDL_PROPAGATION);
+	commands = list_concat(commands,
+						   CreateViewDDLCommandsIdempotent(viewAddress.objectId));
+	commands = lappend(commands, ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
```

Comment:
I noted in a private chat, but probably good to document here as well.

If we use `NodeDDLTaskList` we need `EnsureSequentialMode()`. However, this approach seems to fail for views as `alter_distributed_table`

One another approach could be to use the `metadata connection` for this purpose. Hence, we need to set `ddlJob->metadataSyncCommand` properly

Reply:
I've updated the code to use metadata connection for create or replace view command. Since table type conversion commands do not hit the drop view commands, I've kept them as it is. In other words, I've added `EnsureSequentialMode` to drop view command processor function and used `ddlJob->metadataSyncCommand` properly for create or replace view command processor function.